### DISCUSSION TITLE
getting cmake errors when i try and build a uf2 for my board

### DISCUSSION THREAD

**Discussion by strid3r21** [adafruit/tinyuf2#235] [Q&A] [answered]
these are the notable errors it spits out and i cant make sense of any of it.

`CMake Error at CMakeLists.txt:37 (string):
  string sub-command REPLACE requires at least four arguments.

CMake Error at CMakeLists.txt:38 (string):
  string sub-command STRIP requires two arguments.`

here is the full output of attempted make.

`C:\Users\ubers\Desktop\tinyUF2\ports\espressif>make BOARD=bee_s3 all
idf.py -B_build/bee_s3 -DBOARD=bee_s3 build
Executing action: all (aliases: build)
Running cmake in directory c:\users\ubers\desktop\tinyuf2\ports\espressif\_build\bee_s3
Executing "cmake -G Ninja -DPYTHON_DEPS_CHECKED=1 -DESP_PLATFORM=1 -DBOARD=bee_s3 -DCCACHE_ENABLE=1 c:\users\ubers\desktop\tinyuf2\ports\espressif"...
-- Found Git: C:/Program Files/Git/cmd/git.exe (found version "2.37.1.windows.1")
CMake Error at CMakeLists.txt:37 (string):
  string sub-command REPLACE requires at least four arguments.

CMake Error at CMakeLists.txt:38 (string):
  string sub-command STRIP requires two arguments.

-- ccache will be used for faster recompilation
-- The C compiler identification is GNU 8.4.0
-- The CXX compiler identification is GNU 8.4.0
-- The ASM compiler identification is GNU
-- Found assembler: C:/Espressif/tools/xtensa-esp32s3-elf/esp-2021r2-8.4.0/xtensa-esp32s3-elf/bin/xtensa-esp32s3-elf-gcc.exe
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Espressif/tools/xtensa-esp32s3-elf/esp-2021r2-8.4.0/xtensa-esp32s3-elf/bin/xtensa-esp32s3-elf-gcc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Espressif/tools/xtensa-esp32s3-elf/esp-2021r2-8.4.0/xtensa-esp32s3-elf/bin/xtensa-esp32s3-elf-g++.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Building ESP-IDF components for target esp32s3
-- Project sdkconfig file C:/Users/ubers/Desktop/tinyUF2/ports/espressif/_build/bee_s3/sdkconfig
Loading defaults file C:/Users/ubers/Desktop/tinyUF2/ports/espressif/sdkconfig.defaults...
C:/Users/ubers/Desktop/tinyUF2/ports/espressif/sdkconfig.defaults:21 CONFIG_FLASHMODE_QIO was replaced with CONFIG_ESPTOOLPY_FLASHMODE_QIO
Loading defaults file C:/Users/ubers/Desktop/tinyUF2/ports/espressif/boards/bee_s3/sdkconfig...
-- Found PythonInterp: C:/Espressif/python_env/idf4.4_py3.10_env/Scripts/python.exe (found version "3.10.5")
-- Found Perl: C:/Strawberry/perl/bin/perl.exe (found version "5.32.1")
-- App "tinyuf2" version: 0.10.2-25-gf1d5c36
-- Adding linker script C:/Users/ubers/Desktop/tinyUF2/ports/espressif/_build/bee_s3/esp-idf/esp_system/ld/memory.ld
-- Adding linker script C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_system/ld/esp32s3/sections.ld.in
-- Adding linker script C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.ld
-- Adding linker script C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.api.ld
-- Adding linker script C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.libgcc.ld
-- Adding linker script C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.newlib.ld
-- Adding linker script C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.version.ld
-- Adding linker script C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.newlib-nano.ld
-- Adding linker script C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/soc/esp32s3/ld/esp32s3.peripherals.ld
-- Components: app_trace app_update asio boards bootloader bootloader_support bt cbor cmock coap console cxx driver efuse esp-tls esp32s3 esp_adc_cal esp_common esp_eth esp_event esp_gdbstub esp_hid esp_http_client esp_http_server esp_https_ota esp_https_server esp_hw_support esp_ipc esp_lcd esp_local_ctrl esp_netif esp_phy esp_pm esp_ringbuf esp_rom esp_serial_slave_link esp_system esp_timer esp_websocket_client esp_wifi espcoredump esptool_py expat fatfs freemodbus freertos hal heap idf_test ieee802154 jsmn json lcd led_strip libsodium log lwip mbedtls mdns mqtt newlib nghttp nvs_flash openssl openthread partition_table perfmon protobuf-c protocomm pthread sdmmc soc spi_flash spiffs src tcp_transport tcpip_adapter tinyusb ulp unity usb vfs wear_levelling wifi_provisioning wpa_supplicant xtensa
-- Component paths: C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/app_trace C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/app_update C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/asio C:/Users/ubers/Desktop/tinyUF2/ports/espressif/boards C:/Users/ubers/Desktop/tinyUF2/ports/espressif/components/bootloader C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/bootloader_support C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/bt C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/cbor C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/cmock C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/coap C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/console C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/cxx C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/driver C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/efuse C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp-tls C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp32s3 C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_adc_cal C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_common C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_eth C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_event C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_gdbstub C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_hid C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_http_client C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_http_server C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_https_ota C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_https_server C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_hw_support C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_ipc C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_lcd C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_local_ctrl C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_netif C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_phy C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_pm C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_ringbuf C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_rom C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_serial_slave_link C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_system C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_timer C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_websocket_client C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esp_wifi C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/espcoredump C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/esptool_py C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/expat C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/fatfs C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/freemodbus C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/freertos C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/hal C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/heap C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/idf_test C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/ieee802154 C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/jsmn C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/json C:/Users/ubers/Desktop/tinyUF2/ports/espressif/components/lcd C:/Users/ubers/Desktop/tinyUF2/ports/espressif/components/led_strip C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/libsodium C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/log C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/lwip C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/mbedtls C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/mdns C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/mqtt C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/newlib C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/nghttp C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/nvs_flash C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/openssl C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/openthread C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/partition_table C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/perfmon C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/protobuf-c C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/protocomm C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/pthread C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/sdmmc C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/soc C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/spi_flash C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/spiffs C:/Users/ubers/Desktop/tinyUF2/src C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/tcp_transport C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/tcpip_adapter C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/tinyusb C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/ulp C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/unity C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/usb C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/vfs C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/wear_levelling C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/wifi_provisioning C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/wpa_supplicant C:/Users/ubers/Desktop/tinyUF2/lib/esp-idf/components/xtensa
-- Configuring incomplete, errors occurred!
See also "C:/Users/ubers/Desktop/tinyUF2/ports/espressif/_build/bee_s3/CMakeFiles/CMakeOutput.log".
cmake failed with exit code 1
make: *** [Makefile:31: all] Error 2`

**Comment by hathach** [comment]
what is your IDF hash/version, if it is not the same as the one in the lib/, try to checkout it out first.

**Reply by strid3r21** [reply]
ive got IDF v. 4.4.1 installed

edit: i uninstalled 4.4.1 and installed the idf version that is in lib/ and im still getting the same errors. :(

**Comment by hathach** [comment]
bee_s3 is not one of the supported boards, are you adding new boards ? and are you able to build one of the supported boards ? If yes, then it is probably issue in porting, try to comment/uncomment 1 line each time, and better to based your port on of of the supported boards (which build).

**Reply by strid3r21** [reply]
Yes I was trying to add a new board. I had tried building an already supported board as well and it failed with the same error and I had copied one of the supported boards and changed the relevant information for my board.

I'll try the line by line change and see what happens.

If that doesn't work I think I'll try this on a different PC. Maybe something is screwed up with the setup on this PC.

**Comment by strid3r21** [comment, marked as answer]
i figured it out. i ended up just commenting out the two lines in the CMakeList file that were throwing the error and then it compiled just fine and i was able to flash my board with the new firmware. no idea why those two lines were throwing the error or why they werent needed, but hey got it working. 

lines in question i commented out.

`#string(REPLACE ../../../../lib/ "" GIT_SUBMODULE_VERSIONS ${GIT_SUBMODULE_VERSIONS})  `
`#string(STRIP ${GIT_SUBMODULE_VERSIONS} GIT_SUBMODULE_VERSIONS)`

**Reply by hathach** [reply]
you probably need to run 

```c
git submodule update --init
```

If you done it already, it is probably due to the path difference between posix vs windows.
